proxy/frame_patch_test.go

```diff
@@ -0,0 +1,149 @@
+package proxy
+
+import (
+	"encoding/binary"
+	"github.com/datastax/go-cassandra-native-protocol/primitive"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestPatchQueryConsistency(t *testing.T) {
+	t.Run("valid QUERY body", func(t *testing.T) {
+		query := []byte("SELECT * FROM users;")
+		queryLen := uint32(len(query))
+
+		body := make([]byte, 4+len(query)+2)
+		binary.BigEndian.PutUint32(body[0:4], queryLen)
+		copy(body[4:], query)
+		binary.BigEndian.PutUint16(body[4+len(query):], uint16(primitive.ConsistencyLevelOne))
+
+		err := patchQueryConsistency(body, primitive.ConsistencyLevelQuorum)
+		assert.NoError(t, err)
+
+		offset := 4 + len(query)
+		got := binary.BigEndian.Uint16(body[offset : offset+2])
+		assert.Equal(t, uint16(primitive.ConsistencyLevelQuorum), got)
+	})
+
+	t.Run("too short body", func(t *testing.T) {
+		err := patchQueryConsistency([]byte{0x01, 0x02}, primitive.ConsistencyLevelQuorum)
+		assert.Error(t, err)
+	})
+
+	t.Run("not enough space for consistency", func(t *testing.T) {
+		body := make([]byte, 4)
+		binary.BigEndian.PutUint32(body[0:4], 10)
+		err := patchQueryConsistency(body, primitive.ConsistencyLevelQuorum)
+		assert.Error(t, err)
+	})
+}
+
+func TestPatchExecuteConsistency(t *testing.T) {
+	t.Run("valid EXECUTE body", func(t *testing.T) {
+		id := []byte{0xCA, 0xFE, 0xBA, 0xBE}
+		idLen := len(id)
+
+		body := make([]byte, 2+idLen+2)
+		binary.BigEndian.PutUint16(body[0:2], uint16(idLen))
+		copy(body[2:], id)
+		binary.BigEndian.PutUint16(body[2+idLen:], uint16(primitive.ConsistencyLevelOne))
+
+		err := patchExecuteConsistency(body, primitive.ConsistencyLevelLocalQuorum)
+		assert.NoError(t, err)
+
+		offset := 2 + idLen
+		got := binary.BigEndian.Uint16(body[offset : offset+2])
+		assert.Equal(t, uint16(primitive.ConsistencyLevelLocalQuorum), got)
+	})
+
+	t.Run("too short body", func(t *testing.T) {
+		err := patchExecuteConsistency([]byte{0x00}, primitive.ConsistencyLevelOne)
+		assert.Error(t, err)
+	})
+
+	t.Run("not enough space for consistency", func(t *testing.T) {
+		body := make([]byte, 2)
+		binary.BigEndian.PutUint16(body[0:2], 4)
+		err := patchExecuteConsistency(body, primitive.ConsistencyLevelOne)
+		assert.Error(t, err)
+	})
+}
+
+func TestPatchBatchConsistency(t *testing.T) {
+	tests := []struct {
```

Comment:
Could this use the builtin codec and batch message type? It's verifies more by testing against the reference implementation.

Reply:
See above for the query codec example.